distributed_training/utils/uids.py

```diff
@@ -114,27 +114,25 @@ async def get_random_uids(
 
 
 async def map_uid_to_peerid(self, uids):
+    # Get all peers connected to our DHT, their ips and their ports
+    peer_list_dht = await self._p2p.list_peers()
+    peer_list_dht_addrs = [
+        str(peer.addrs[0]).split("/ip4/")[1].split("/")[0] for peer in peer_list_dht
+    ]
+    peer_list_dht_ports = [str(peer.addrs[0]).split("/")[-1] for peer in peer_list_dht]
+
+    # Get only peers connected to the current run id
+    prefix = self.grad_averager.matchmaking_kwargs["prefix"]
+    metadata, _ = self.dht.get(f"{prefix}.all_averagers", latest=True) or (
+        {},
+        None,
+    )
+
```

Comment:
@mikkelmedm these where the optimisations made to `map_uid_to_peerid`. The extraction of the peer list and their addresses where moved outside of the for loop since they didn't need to be run more than once.